pallet/staking/src/lib.rs

```diff
@@ -867,14 +819,14 @@ pub mod pallet {
 					}
 				}
 
-				if T::RingCurrency::deposit_into_existing(&c, c_payout).is_ok() {
+				if T::OnSessionEnd::reward(&c, c_payout).is_ok() {
```

Comment:
Suggest adding another event to track the cases of reward failures. One of the reasons after this PR is that the treasury fund has run out. Catch the error in the pallet instead of the `reward` function to avoid duplicate error handling.